src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Date.cs

```diff
@@ -37,13 +37,23 @@ public static bool TryFormat(DateTimeOffset value, Span<byte> destination, out i
                 offset = value.Offset;
             }
 
-            return symbol switch
+            switch (symbol)
             {
-                'R' => DateTimeFormat.TryFormatR(value.UtcDateTime, new TimeSpan(DateTimeFormat.NullOffset), destination, out bytesWritten),
-                'l' => TryFormatDateTimeL(value.UtcDateTime, destination, out bytesWritten),
-                'O' => DateTimeFormat.TryFormatO(value.DateTime, value.Offset, destination, out bytesWritten),
-                'G' => TryFormatDateTimeG(value.DateTime, offset, destination, out bytesWritten),
-                _ => FormattingHelpers.TryFormatThrowFormatException(out bytesWritten),
+                case 'R':
+                    return DateTimeFormat.TryFormatR(value.UtcDateTime, new TimeSpan(DateTimeFormat.NullOffset), destination, out bytesWritten);
+
+                case 'O':
+                    return DateTimeFormat.TryFormatO(value.DateTime, value.Offset, destination, out bytesWritten);
+
+                case 'l':
+                    return TryFormatDateTimeL(value.UtcDateTime, destination, out bytesWritten);
+
+                case 'G':
+                    return TryFormatDateTimeG(value.DateTime, offset, destination, out bytesWritten);
+
+                default:
+                    ThrowHelper.ThrowFormatException_BadFormatSpecifier();
```

Comment:
Should Roslyn recognize DoesNotReturns in this throw helper so you don't need the bogus goto after it?

Reply:
Roslyn only uses `DoesNotReturn` as part of definite assignment analysis, such as to support nullable reference types.

It doesn't impact reachability for control flow purposes or what IL actually has to be emitted to keep the JIT happy.